src/main/groovy/com/github/spotbugs/snom/internal/SpotBugsTaskFactory.java

```diff
@@ -63,30 +68,35 @@ private void generateForJava(Project project, Action<? super SpotBugsTask> confi
 
   private void generateForAndroid(
       Project project, Action<? super SpotBugsTask> configurationAction) {
-    project
-        .getPlugins()
-        .withId(
-            "com.android.application",
-            plugin ->
-                project
-                    .getTasks()
-                    .withType(AndroidJavaCompile.class)
-                    .all(
-                        task -> {
-                          String name = GUtil.toLowerCamelCase("spotbugs " + task.getVariantName());
-                          log.debug("Creating SpotBugsTask for {}", task);
-                          project
-                              .getTasks()
-                              .register(
-                                  name,
-                                  SpotBugsTask.class,
-                                  spotbugsTask -> {
-                                    spotbugsTask.setSourceDirs(task.getSource());
-                                    spotbugsTask.setClassDirs(
-                                        task.getOutputDirectory().getAsFileTree());
-                                    spotbugsTask.setAuxClassPaths(task.getClasspath());
-                                    configurationAction.execute(spotbugsTask);
-                                  });
-                        }));
+    if (project.getPlugins().hasPlugin("com.android.base")) {
```

Comment:
better to use [the `withId()` method](https://docs.gradle.org/5.6/javadoc/org/gradle/api/plugins/PluginContainer.html#withId-java.lang.String-org.gradle.api.Action-), then we do not need to care the order of loading plugin.

Reply:
Although I would agree with you generally, it unfortunately is not that simple for the AGP. If you use `withId()` on the base plugin, it will try to evaluate the block before the actual application/library plugin has applied.

That being said, I agree that it's better to find a solution where the loading order is not relevant, so then I need to look for more specific AGP plugins.